RecoEgamma/EgammaTools/plugins/EGPfIsolationModifier.cc

```diff
@@ -0,0 +1,269 @@
+#include "CommonTools/CandAlgos/interface/ModifyObjectValueBase.h"
+
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "DataFormats/Common/interface/ValueMap.h"
+
+#include "DataFormats/EgammaCandidates/interface/GsfElectron.h"
+#include "DataFormats/EgammaCandidates/interface/Photon.h"
+
+namespace {
+  const edm::EDGetTokenT<edm::ValueMap<float> > empty_token;
+  const static edm::InputTag empty_tag("");
+}
+
+#include <unordered_map>
+
+class EGPfIsolationModifierFromValueMaps : public ModifyObjectValueBase {
+public:
+  struct electron_config {
+    edm::InputTag electron_src;
+    edm::InputTag sumChargedHadronPt;
+    edm::InputTag sumNeutralHadronEt;
+    edm::InputTag sumPhotonEt;
+    edm::InputTag sumChargedParticlePt;
+    edm::InputTag sumNeutralHadronEtHighThreshold;
+    edm::InputTag sumPhotonEtHighThreshold;
+    edm::InputTag sumPUPt;
+    edm::EDGetTokenT<edm::View<pat::Electron> > tok_electron_src;
```

Comment:
(I made similar comments to Matteo): these tag,token, stringName for the slew of valueMaps can be combined in a small struct, which can then be stuffed into a vector and more easily manipulated than the current verbose and repetitive version